### DISCUSSION TITLE
Dicas e referências para o aprendizado React em 2019?

### DISCUSSION THREAD

**Discussion by mikaelhadler** [frontendbr/forum#1592] [Perguntas e Respostas]
Olá pessoal, hoje vim pedir dicas sobre @reactjs

Minhas principais dúvidas são:

- Quais formas de organizar sua arvore de componentes de forma que facilite uma boa estratégia de controle de estado?
- Quais foram suas principais dificuldades ?
- Recomendam algum roadmap ?
- Quais cursos recomendam ?
- Quais principais UI Frameworks vocês utilizavam ?
- Quais artigos vocês acharam essenciais para o aprendizado ?
- Quais pessoas vocês seguem no Twitter que postam um conteúdo bacana sobre ?

**Comment by talitaazevedo** [comment]
@mikaelhadler ,
Bom dia! 
Eu estou estudando muito esta tecnologia,  essa lib é realmente sensacional, se estiver procurando um curso ou algo do tipo, recomendo fortemente o curso  https://www.devpleno.com/devreactjs/ do Devpleno, é bem completo e explicativo.

Abaixo alguns links  que considerei bem interessantes sobre está tecnologia.

https://medium.com/@alexmngn/how-to-better-organize-your-react-applications-2fd3ea1920f1

https://reactpatterns.com/

https://medium.com/@franleplant/react-higher-order-components-in-depth-cf9032ee6c3e

https://developmentarc.gitbooks.io/react-indepth

**Comment by wilsonneto-swe** [comment]
Cara, 
quanto a roadmap pensando em front indico:
- Html
- Css
- JS
- Aprofundar em ES 6 ~ 10
- Saas
- Bootstrap / outras libs visuais
- ReactJS e VueJs também.

Sobre cursos um que me ajudou demais foi o do devpleno:
[http://bit.ly/reactjs-dev-pleno](http://bit.ly/reactjs-dev-pleno)

E vá com calma, aprendendo com dedicação e capricho.

Abraços!

**Comment by romulosardinha** [comment]
Acho que um pessoal que está gerando um bom material é a https://rocketseat.com.br/

**Comment by ninetails** [comment]
Nomes a seguir:
- [Dan Abramov](https://twitter.com/dan_abramov) ([Blog](https://overreacted.io/))
- [Andrew Clark](https://twitter.com/acdlite)
- [Kent C. Dodds](https://kentcdodds.com/)

Os dois primeiros são do **core team**, o primeiro foi criador do _Redux_. O segundo é bem famoso por causa da _testing-library_, reeducando testes para frontend e já fez bastante coisa enquanto trampava no PayPal. Cursos com o Dan ou o Kent são muito bons. Recomendo também o [Frontend Masters](https://frontendmasters.com/), tem muita coisa legal por lá e de gente foda.

Pra ficar atualizado, o [Blog oficial](https://reactjs.org/blog/) é bem foda, rola muita comunicação por [coisas que há por vir](https://github.com/reactjs/rfcs).

Pra comunidades, tem o [Slack do React Brasil](https://react-brasil-slack.herokuapp.com/) que inclusive tem [meetups](http://meetu.ps/c/2DvKV/jBQxM/d) caso resida em SP.

Já pra questão de estrutura de arquivos, tem um ótimo guia do Dan Abramov: https://react-file-structure.surge.sh/

**Comment by wilsonneto-swe** [comment]
Um curso muito bom que indico para quem está começando, e que abriu as inscrições ontem, é o Fullstack do Dev Pleno que aborda React, NodeJs e ReactNative com projetos do zero até o deploy.

A pegada dele é para quem está no início, ainda não tem muita experiência e quer já aprender alinhado com estas tecnologias e com o mercado.

Link do curso: [Curso FullStack - DevPleno](http://bit.ly/fullstack-dev-pleno)

Espero que ajude, abraços!

**Comment by mikaelhadler** [comment]
> @mikaelhadler ,
> Bom dia!
> Eu estou estudando muito esta tecnologia, essa lib é realmente sensacional, se estiver procurando um curso ou algo do tipo, recomendo fortemente o curso https://www.devpleno.com/devreactjs/ do Devpleno, é bem completo e explicativo.
> 
> Abaixo alguns links que considerei bem interessantes sobre está tecnologia.
> 
> https://medium.com/@alexmngn/how-to-better-organize-your-react-applications-2fd3ea1920f1
> 
> https://reactpatterns.com/
> 
> https://medium.com/@franleplant/react-higher-order-components-in-depth-cf9032ee6c3e
> 
> https://developmentarc.gitbooks.io/react-indepth

Muito obrigado pelas dicas, estão sendo muito úteis :3

**Comment by mikaelhadler** [comment]
> Cara,
> quanto a roadmap pensando em front indico:
> 
>     * Html
> 
>     * Css
> 
>     * JS
> 
>     * Aprofundar em ES 6 ~ 10
> 
>     * Saas
> 
>     * Bootstrap / outras libs visuais
> 
>     * ReactJS e VueJs também.
> 
> 
> Sobre cursos um que me ajudou demais foi o do devpleno:
> http://bit.ly/reactjs-dev-pleno
> 
> E vá com calma, aprendendo com dedicação e capricho.
> 
> Abraços!

Valeu pela dica, conhecia o curso do dev pleno, porém não ta rolando grana pro aporte, mas vai rolar. Me parece um overview bem completo.

**Comment by mikaelhadler** [comment]
> Nomes a seguir:
> 
>     * [Dan Abramov](https://twitter.com/dan_abramov) ([Blog](https://overreacted.io/))
> 
>     * [Andrew Clark](https://twitter.com/acdlite)
> 
>     * [Kent C. Dodds](https://kentcdodds.com/)
> 
> 
> Os dois primeiros são do **core team**, o primeiro foi criador do _Redux_. O segundo é bem famoso por causa da _testing-library_, reeducando testes para frontend e já fez bastante coisa enquanto trampava no PayPal. Cursos com o Dan ou o Kent são muito bons. Recomendo também o [Frontend Masters](https://frontendmasters.com/), tem muita coisa legal por lá e de gente foda.
> 
> Pra ficar atualizado, o [Blog oficial](https://reactjs.org/blog/) é bem foda, rola muita comunicação por [coisas que há por vir](https://github.com/reactjs/rfcs).
> 
> Pra comunidades, tem o [Slack do React Brasil](https://react-brasil-slack.herokuapp.com/) que inclusive tem [meetups](http://meetu.ps/c/2DvKV/jBQxM/d) caso resida em SP.
> 
> Já pra questão de estrutura de arquivos, tem um ótimo guia do Dan Abramov: https://react-file-structure.surge.sh/

Irado! Conhecia o Dan. Abramov e o Kent C. Dodds, valeu pela dica. Os cursos deles são realmente muito bons, consegui uns gratuitos pelo egghead.io e já ta bem da hora a evolução. 

Confesso que to tendo bastante dificuldade de pegar um workflow com o react, é tão diferente do Vue, mas to achando muito foda. To pegando uns códigos bem limpos e intuitivos, então ta só a evolução.

**Comment by mikaelhadler** [comment]
> Acho que um pessoal que está gerando um bom material é a https://rocketseat.com.br/

Valeu pela dica = )

**Comment by coelhucas** [comment]
Sugiro um roadmap similar ao que o @wilsonneto-dev falou. Antes de partir para o React é bom ter uma noção forte do conceitos mais básicos, domínio do JS, HTML 5, CSS. Estudar um pouco UX e pegar referências é uma boa também.

Sobre cursos, eu sugiro evitar, ao menos de começo. Você vai receber uma enxurrada de bibliotecas (o que é legal de se conhecer) e métodos específicos de resolver um problema com soluções que são muito além do pequeno problema que você tinha a princípio, aprendendo "por conta", você não só vai começar a desenvolver melhor seu próprio pensamento crítico quanto poderá pensar em como seria a solução de um problema simples sem precisar sempre de uma biblioteca para.
